[ADD] Maintenance: Maintenance requests (#8139)
odoo/documentation: content/applications/inventory_and_mrp/maintenance/maintenance_requests.rst

====================
Maintenance requests
====================

.. |MO| replace:: :abbr:`MO (Manufacturing Order)`

In order to keep equipment and work centers functioning properly, it is often necessary to perform
maintenance on them. This can include preventive maintenance, intended to prevent equipment from
breaking down, or corrective maintenance, which is used to fix equipment that is broken or otherwise
unusable.

In Odoo *Maintenance*, users can create *maintenance requests* to schedule and track the progress of
equipment and work center maintenance.

Create maintenance request
==========================

> To create a new maintenance request, navigate to :menuselection:`Maintenance --> Maintenance -->

Reviewer: ```suggestion
To create a new maintenance request, navigate to :menuselection:`Maintenance app --> Maintenance -->
```
Just going off the way I've been instructed to write it

> To create a new maintenance request, navigate to :menuselection:`Maintenance --> Maintenance -->

Reviewer 2: ```suggestion
To create a new maintenance request, navigate to :menuselection:`Maintenance app--> Maintenance -->
```
yeah, adding 'app' seems to be the standard use here

> Maintenance Requests`, and then click :guilabel:`New`.

Reviewer: ```suggestion
Maintenance Requests`, and click :guilabel:`New`.
```

> Begin filling out the form by entering a descriptive title in the :guilabel:`Request` field (e.g.

Reviewer: ```suggestion
Begin filling out the form by entering a descriptive title in the :guilabel:`Request` field (e.g.,
```

> "Drill not working").

Reviewer 2: ```suggestion
'Drill not working').
```

> 

Reviewer 2: may want to add a quick mention about the 'Created By' field -- just for a more complete overview of the form

In the :guilabel:`For` drop-down menu, select :guilabel:`Equipment` if the maintenance request is
being created for a piece of equipment, or :guilabel:`Work Center` if it is being created for a work
center.

Depending on the option selected in the :guilabel:`For` field, the next field is titled either
:guilabel:`Equipment` or :guilabel:`Work Center`. Using the drop-down menu for either field, select
a piece of equipment or a work center.

> 

Reviewer 2: I'm also seeing a 'Worksheet Template' field in the runbot I'm using -- is this present in most dbs or is this runbot a bit "funky"? haha worth a quick double-check either way

The next field is titled :guilabel:`Request Date`, and is set by default to the date on which the
maintenance request is created. This date cannot be changed by the user.

> In the :guilabel:`Maintenance` field, select the :guilabel:`Corrective` option if the request is

Reviewer 2: ```suggestion
In the :guilabel:`Maintenance Type` field, select the :guilabel:`Corrective` option if the request is
```

intended to fix an existing issue, or the :guilabel:`Preventive` option if the request is intended
to prevent issues from occurring in the future.

If the request is being created to address an issue that arose during a specific manufacturing order
(MO), select it in the :guilabel:`Manufacturing Order` field.

If an |MO| was selected in the :guilabel:`Manufacturing Order` field, a :guilabel:`Work Order` field
appears below it. If the issue arose during a specific work order, specify it in this field.

In the :guilabel:`Team` field, select the maintenance team that is responsible for managing the
request. If a specific team member is responsible, select them in the :guilabel:`Responsible` field.

The :guilabel:`Scheduled Date` field is used to specify the date on which maintenance should take
place, and the time it should begin. Choose a date by clicking on the field to open a calendar in a
pop-up window, and then select a day on the calendar. Enter an hour and minute in the two fields

> below the calendar, and then click :guilabel:`Apply` to save the date and time.

Reviewer: ```suggestion
below the calendar, and click :guilabel:`Apply` to save the date and time.
```

The :guilabel:`Duration` field is used to specify the time it takes to complete the maintenance
request. Use the text-entry field to enter the time in a `00:00` format.

If :guilabel:`Work Center` was selected in the :guilabel:`For` field, a :guilabel:`Block Workcenter`

> checkbox appears below the :guilabel:`Duration` field. Enable the checkbox to prevent work orders or
> other maintenance from being scheduled at the specified work center while the maintenance request is

Reviewer 2: ```suggestion
checkbox appears below the :guilabel:`Duration` field. Enable the checkbox to prevent work orders, or
other maintenance, from being scheduled at the specified work center, while the maintenance request is
```

being processed.

> The :guilabel:`Priority` field is used to communicate the importance or urgency of the maintenance

Reviewer 2: ```suggestion
The :guilabel:`Priority` field is used to communicate the importance (or urgency) of the maintenance
```

request. Assign the request a priority between zero and three stars, by clicking on the desired star

> number. Requests assigned a higher priority appear above those with a lower priority, on the Kanban

Reviewer 2: this might be an opportunity to use a star emoji -- what do you think?

> request. Assign the request a priority between zero and three stars, by clicking on the desired star
> number. Requests assigned a higher priority appear above those with a lower priority, on the Kanban

Reviewer 2: ```suggestion
request. Assign the request a priority between zero and three stars, by clicking on the desired star
number. Requests assigned a higher priority appear above those with a lower priority on the Kanban
```

board used to track the progression of maintenance requests.

In the :guilabel:`Notes` tab at the bottom of the form, enter any relevant details about the

> maintenance request (why the maintenance issue arose, when it occurred, etc.).

Reviewer: ```suggestion
maintenance request, such as why the maintenance issue arose, when it occurred, etc.
```

The :guilabel:`Instructions` tab is used to include instructions for how maintenance should be

> performed. Select one of the three options, and then include the instructions as detailed below:

Reviewer: ```suggestion
performed. Select one of the three options, and include the instructions as detailed below:
```

- :guilabel:`PDF`: click the :guilabel:`Upload your file` button to open the device's file manager,

>   and then select a file to upload.

Reviewer: ```suggestion
  and select a file to upload.
```

> - :guilabel:`Google Slide`: enter a Google Slide link in the text-entry field that appears after the

Reviewer 2: ```suggestion
- :guilabel:`Google Slide`: enter a :guilabel:`Google Slide link` in the text-entry field that appears after the
```
using a GUI here because that's how it actually appears on the UI

  option is selected.
- :guilabel:`Text`: enter the instructions in the text-entry field that appears after the option is
  selected.

.. image:: maintenance_requests/request-form.png
   :align: center
   :alt: A maintenance request form filled out for a piece of equipment.

> Process maintenance request

Reviewer: I think an additional screenshot in this short section would be nice to have. Potentially one that shows the Maintenance request pipeline - the different stages, a few tasks in each stage, etc. - up to you

===========================

Once a maintenance request has been created, it appears in the :guilabel:`New Request` stage of the
:guilabel:`Maintenance Requests` page, which can be accessed by navigating to

> :menuselection:`Maintenance --> Maintenance --> Maintenance Requests`.

Reviewer 2: ```suggestion
:menuselection:`Maintenance app --> Maintenance --> Maintenance Requests`.
```

> :menuselection:`Maintenance --> Maintenance --> Maintenance Requests`.

Reviewer: ```suggestion
:menuselection:`Maintenance app --> Maintenance --> Maintenance Requests`.
```

Maintenance requests can be moved to different stages by dragging and dropping them. They can also
be moved by clicking on a request to open it in a new page, and then selecting the desired stage

> from the stage indicator bar above the top-right corner of the request's form.

Reviewer 2: ```suggestion
from the stage indicator bar, located above the top-right corner of the request form.
```

Successful maintenance requests should be moved to the :guilabel:`Repaired` stage, indicating that

> the specified piece of equipment or work center is repaired.

Reviewer 2: ```suggestion
the specified piece of equipment, or work center, is repaired.
```

> Failed maintenance requests should be moved to the :guilabel:`Scrap` stage, indicating the the

Reviewer: ```suggestion
Failed maintenance requests should be moved to the :guilabel:`Scrap` stage, indicating the
```
